cpp/include/cudf/io/json.hpp

```diff
@@ -508,6 +526,19 @@ class json_reader_options_builder {
     return *this;
   }
 
+  /**
+   * @brief Set whether to prune columns only to be read using dtypes as input columns.
+   * This option is useful for parsing only a subset of columns.
```

Comment:
*nit*: I don't think I understand what this means. I _think_ this means that if one sets the `dtypes` via the `std::map` interfaces, then columns whose names are not in the keys of that map are not returned?

What about the case when one uses `dtypes(std::vector<data_type> dtypes)`?

Reply:
dtypes can be given as schema, map, and vector. For both schema and map, it will work as you mentioned.
For `std::vector<data_type> dtypes`, it will return first `n` columns from the input dtypes. There is one unit test in https://github.com/rapidsai/cudf/pull/14996/files#diff-372701f46ff1250e85be64b010783d16dbb0826f2845a412a482a1ea4ddfa023R2438-R2447 where it's tested.

Reply:
OK, how about:
```
@brief Set whether to prune columns on read, selected based on the @p dtypes option.

When @p prune_columns is true, if the reader options include @p dtypes, then
the reader will only return those columns which are mentioned in @p dtypes.
If false, then all columns are returned, independent of the @p dtypes setting.

@param val Boolean value to enable/disable column pruning
@return this for chaining
```

(I may have the doxygen formatting wrong)